sdk-core/src/main/java/care/data4life/sdk/network/model/EncryptedKey.kt

```diff
@@ -13,34 +13,6 @@
  * applications and/or if you’d like to contribute to the development of the SDK, please
  * contact D4L by email to help@data4life.care.
  */
+package care.data4life.sdk.network.model
 
-package care.data4life.sdk.network.model;
-
-import java.util.Objects;
-
-public class EncryptedKey {
-
-    private String encryptedKeyBase64;
-
-
-    public EncryptedKey(String encryptedKeyBase64) {
-        this.encryptedKeyBase64 = encryptedKeyBase64;
-    }
-
-    public String getEncryptedKey() {
-        return encryptedKeyBase64;
-    }
-
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        EncryptedKey that = (EncryptedKey) o;
-        return Objects.equals(encryptedKeyBase64, that.encryptedKeyBase64);
-    }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(encryptedKeyBase64);
-    }
-}
+data class EncryptedKey(val encryptedKey: String)
```

Comment:
the power of Kotlin 🥇

Reply:
maybe the name could reflect the encoding `encryptedKey` -> `base64Key` or `keyBase64`

don't know if it would be possible to enforce base64 encoding with a check here

Reply:
A check is a bit difficult, since it is not really dividable from a normal string. But we can put a factory on top (since it is not really feasible to make it happen with Kotlins constructors in data classes) which does the actual encoding. That means we can pull this out of the CryptoService (which should not know something about encoding in my opinion). Also we might then also want to a `asDecoded` method/getter to this class (to pull the decoding also out of the CryptoService. What do you think?

Reply:
👍